### ISSUE TITLE
`install_github` with non-default host broken: "attempt to set an attribute on NULL"

### ISSUE THREAD

**Issue by r-cheologist** [r-lib/remotes#396]
Consider the following:

    requireNamespace("remotes")
    remotes::install_github(
      repo = 'loosolab/MARMoSET', host = 'github.molgen.mpg.de/api/v3')

This fails with the error:

    Error: Failed to install 'MARMoSET' from GitHub:
      attempt to set an attribute on NULL

This happens both with the current cran version and after invoking `remotes::install_github("r-lib/remotes")`.

**Comment by dpprdan** [comment]
The error message is caused by `ratelimit_reset <- .POSIXct(res_headers$`x-ratelimit-reset`, tz = "UTC")` in `github_error` since this server apparently does not return a `x-ratelimit-reset` in its header, so ``res_headers$`x-ratelimit-reset`` = NULL`. I gather that for this server the REST API rate limit is not set. In any case, this seems to be possible according to the docs 

> Note that the limits mentioned above are the default rate limits for a GitHub Enterprise Server instance. Contact your site administrator to confirm _if rate limits are enabled_ and how they are configured.

Emphasis mine. 

With my attempt to fix this applied (#413), we now get

``` r
remotes::install_github(
  repo = "loosolab/MARMoSET", host = "github.molgen.mpg.de/api/v3"
)
#> Error: Failed to install 'unknown package' from GitHub:
#>   HTTP error 401.
#>   Bad credentials
```

This is because `install_github()` uses the `auth_token = github_pat()` default.

Unfortunately `auth_token = NULL` does not help (yet).

``` r
remotes::install_github(
  repo = "loosolab/MARMoSET", host = "github.molgen.mpg.de/api/v3", 
  auth_token = NULL
)
#> Error: Failed to install 'unknown package' from GitHub:
#>   HTTP error 401.
#>   Bad credentials
```

This is because [`remote_package_name.github_remote`](https://github.com/r-lib/remotes/blob/681da93f01bf727061231c5de89691b790b1a681/R/install-github.R#L211) calls `github_DESCRIPTION()` with [`pat = remote$auth_token %||% github_pat()`](https://github.com/r-lib/remotes/blob/681da93f01bf727061231c5de89691b790b1a681/R/install-github.R#L230). So since `auth_token = NULL`, `github_DESCRIPTION()` essentially tries to access the DESCRIPTION file on the `github.molgen.mpg.de/api/v3` host with the PAT from the `GITHUB_PAT` environment variable (which is likely to be the one for github.com). Since `github.molgen.mpg.de/api/v3` does not require a PAT, we get the 401 error.

Since @jimhester deliberately set `%||% github_pat()` in commit 203d06e, I don't know how to solve this without breaking other things at this moment. 

A workaround for this particular case for the time being is:

```r
Sys.unsetenv("GITHUB_PAT")
remotes::install_github(
  repo = "loosolab/MARMoSET", host = "github.molgen.mpg.de/api/v3"
)
```

**Comment by reikoch** [comment]
I encountered a similar problem:
my GITHUB_PAT environment variable contained the PAT for github.com and that effectively disabled installations from our github enterprise server.

Somehow I think the PATs have to be linked to the corresponding host:
```
    github.com PAT1
    github.example.com PAT2
    ...
```

**Comment by jimhester** [comment]
Fixed by #413
